pkg/nodeconfig/payload/payload.go

```diff
@@ -110,6 +110,36 @@ param(
     }
   }
 
+# This retries getting the HNS-network until it succeeds
+function Retry-GetHnsNetwork {
+     param (
```

Comment:
>        [int]$retryCount = 20,
>         [int]$retryDelay = 10

Do you anticipate a case where the input params are needed? Otherwise, consider defining just local variables instead of params.

Reply:
I was thinking it might be good to have if we want to change them later, for testing or something? IDK.

Reply:
https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it